frontend/test/resolution/testExternBlocks.cpp

```diff
@@ -156,6 +156,124 @@ static void test7() {
          true /* foo comes from the extern block */);
 }
 
+static void test8() {
+  printf("%s\n", __FUNCTION__);
+  {
+    Context* context = buildStdContext();
+    ErrorGuard guard(context);
+    QualifiedType qt =
+      resolveTypeOfXInit(context,
+                         R""""(
+                          module M {
+                            extern { int i; }
+                            var x = i;
+                          }
+                         )"""", true);
+
+    assert(!qt.isUnknownOrErroneous());
+    assert(qt.type()->isIntType());
+
+    assert(!guard.realizeErrors());
+  }
+
+  {
+    Context* context = buildStdContext();
+    ErrorGuard guard(context);
+    QualifiedType qt =
+      resolveTypeOfXInit(context,
+                         R""""(
+                          module M {
+                            extern { }
+                            var x = 1;
+                          }
+                         )"""", true);
+
+    assert(!qt.isUnknownOrErroneous());
+
+    assert(!guard.realizeErrors());
+  }
+}
+
+static void test9() {
+  printf("%s\n", __FUNCTION__);
+  Context* context = buildStdContext();
+  ErrorGuard guard(context);
+  QualifiedType qt =
+    resolveTypeOfXInit(context,
+                       R""""(
+                        module M {
+                          module CDemo {
+                            extern {
+                               static double square(double num) {
+                                  return 2;
+                               }
+                            }
+                          }
+
+                          var x = CDemo.square(2.0);
+                        }
+                       )"""", true);
+
+  assert(!qt.isUnknownOrErroneous());
+  assert(qt.type()->isRealType());
+  assert(qt.type()->toRealType()->isDefaultWidth());
+
+  assert(!guard.realizeErrors());
+}
+
+static void test10() {
+  printf("%s\n", __FUNCTION__);
+  Context* context = buildStdContext();
+  ErrorGuard guard(context);
+  QualifiedType qt =
+    resolveTypeOfXInit(context,
+                       R""""(
+                        module M {
+                          module CDemo {
+                            extern {
+                               int main(void) {
+                                  return 1;
+                               }
+                            }
+                          }
+
+                          var x = CDemo.main();
+                        }
+                       )"""", true);
+
+  assert(!qt.isUnknownOrErroneous());
+  assert(qt.type()->isIntType());
+  assert(!qt.type()->toIntType()->isDefaultWidth());
```

Comment:
This seems like it's almost certainly not right. C++ `int` is 32-bit, no? And, last I checked, it's only guaranteed by implementation to be 16 bit. 64 bit is certainly wider than either of those.

Reply:
I can't read. Thanks!